### PR TITLE
Add getting started Vue.js

### PR DESCRIPTION
**Pull request by berthieresteban**
## What does this PR do?

This getting-started has been added to the documentation before but removed because vuepress can't build with it. 

It was approved in those PR : #405 and #406 

Add VueJS getting started without vuex to SDK JS6 documentation.
Resolve the build problem of the vuepress doc (Vuepress tried to build .vue files of the getting-started and crash) by adding symlink.

image

:warning:  need to add following line in doc/6/getting-started/vuejs/standalone/index.md line 14 after merge: 
``` You can find the full code of this guide [here](https://github.com/kuzzleio/sdk-javascript/tree/6-dev/doc/6/getting-started/vuejs/standalone).```:warning: 

### How should this be manually tested?
See documentation: 
  - Step 1 : npm run doc prepare
  - Step 2 : npm run doc-dev

Test getting started project (You'll need the CYPRESS_RECORD_KEY as environment variable)
  - Step 1 : Get a Kuzzle running without anonymous restrictions
  - Step 2 : cd doc/6/getting-started/vuejs && npm ci
  - Step 3 : npm run serve-standalone
  - Step 4 : npm run test

### Other changes
/

### Boyscout
/

### CODE REVIEW

**Comment by Aschen** [comment]
This PR should be on `master` branch

File: doc/6/getting-started/.vuejs/cypress/plugins/index.js [hunk 1]
```diff
@@ -0,0 +1,17 @@
+// ***********************************************************
```
Comment (scottinet) [inline doc/6/getting-started/.vuejs/cypress/plugins/index.js]: is this file needed?

File: doc/6/getting-started/.vuejs/cypress/support/commands.js [hunk 2]
```diff
@@ -0,0 +1,125 @@
+// ***********************************************
+// This example commands.js shows you how to
```
Comment (scottinet) [inline doc/6/getting-started/.vuejs/cypress/support/commands.js]: Explanation comments should probably be removed.

File: doc/6/getting-started/.vuejs/cypress/support/index.js [hunk 3]
```diff
@@ -0,0 +1,20 @@
+// ***********************************************************
```
Comment (scottinet) [inline doc/6/getting-started/.vuejs/cypress/support/index.js]: is this file needed?

File: doc/6/getting-started/vuejs/standalone/index.md [hunk 4]
```diff
@@ -0,0 +1,118 @@
...
+
+<<< ./snippets/App.vue.snippet:3[js]
+
+Then we will [etablish the connection](/sdk/js/6/core-classes/kuzzle/connect/) to kuzzle and create, if they don't [exists](sdk/js/6/controllers/index/exists/), the [index](sdk/js/6/controllers/index/create/) and [collection](sdk/js/6/controllers/collection/create/) of our chat. 
+Add the following `valid()` method in the export of the `<script>` tag of your _App.vue_ file:
+
+<<< ./snippets/App.vue.snippet:2[js]
+
+## Get the username
+First of all, we need to know the user nickname. Let's start by adding the following html code in the `<template>` tag to allow the client type his nickname:
```
Comment (scottinet) [inline doc/6/getting-started/vuejs/standalone/index.md]: A nickname is a familiar/affective form of a proper name:

```suggestion
First of all, we need to know the user's name. Let's start by adding the following HTML code in the `<template>` tag, allowing one to enter their name:
```

File: doc/6/getting-started/vuejs/standalone/index.md [hunk 5]
```diff
@@ -0,0 +1,118 @@
...
+
+<<< ./snippets/App.vue.snippet:4[js]
+
+Then, create the following functions to fetch and display the messages: 
+
+<<< ./snippets/App.vue.snippet:5[js]
+
+<<< ./snippets/App.vue.snippet:6[js]
+
+The function `fetch_message()` will [search](/sdk/js/6/controllers/document/search/) for the first hundred newest messages and store them in our array before subscribe to notification about `messages` collection. We called it in the `valid()` function we created before.
```
Comment (scottinet) [inline doc/6/getting-started/vuejs/standalone/index.md]:
```suggestion
The function `fetch_message()` will [search](/sdk/js/6/controllers/document/search/) for the first hundred newest messages and store them in our array, before subscribing to changes in the `messages` collection. We called it in the `valid()` function we created above.
```

File: doc/6/getting-started/vuejs/standalone/index.md [hunk 6]
```diff
@@ -0,0 +1,118 @@
...
+
+<<< ./snippets/App.vue.snippet:7[js]
+
+Now, just add the following html code to display the messages:
```
Comment (scottinet) [inline doc/6/getting-started/vuejs/standalone/index.md]: Too many "just" usages: limit it to seemingly complex tasks that can be accomplished very easily by a simple action:

```suggestion
Now, add the following HTML code to display the messages:
```

File: doc/6/getting-started/vuejs/standalone/index.md [hunk 7]
```diff
@@ -0,0 +1,118 @@
...
+
+<<< ./snippets/App.vue.snippet:8[html]
+
+To finish this part, just add the following css classes:
```
Comment (scottinet) [inline doc/6/getting-started/vuejs/standalone/index.md]:
```suggestion
To finish this part, add the following CSS classes:
```

File: doc/6/getting-started/vuejs/standalone/index.md [hunk 8]
```diff
@@ -0,0 +1,118 @@
...
+
+<<< ./snippets/App.vue.snippet:9[css]
+
+We can now display the messages stored in Kuzzle. Cool but we need to create some right ?
```
Comment (scottinet) [inline doc/6/getting-started/vuejs/standalone/index.md]: 1. Stay formal
2. Never put a space before punctuations such as `? ! : ;` 

```suggestion
We can now display the messages stored in Kuzzle. All there is to do is to create some.
```

File: doc/6/getting-started/vuejs/standalone/index.md [hunk 9]
```diff
@@ -0,0 +1,118 @@
...
+
+## Send messages
+
+We need to write a simple method that will [create](/sdk/js/6/controllers/document/create/) a new message document in Kuzzle.
+
+<<< ./snippets/App.vue.snippet:10[js]
+
+As you can see we don't push the new message in our array on message creation.
+Actually, We'll subscribe to the collection that contains our messages.
```
Comment (scottinet) [inline doc/6/getting-started/vuejs/standalone/index.md]: I... think this sentence is erroneous. I'll try to propose a fix, hoping I correctly understood what were your intentions.
Also, don't use the future tense to describe an existing situation.

```suggestion
Actually, we need to subscribe to the collection that contains our messages.
```

File: doc/6/getting-started/vuejs/standalone/index.md [hunk 10]
```diff
@@ -0,0 +1,118 @@
...
+
+We can now display the messages stored in Kuzzle. Cool but we need to create some right ?
+
+## Send messages
+
+We need to write a simple method that will [create](/sdk/js/6/controllers/document/create/) a new message document in Kuzzle.
+
+<<< ./snippets/App.vue.snippet:10[js]
+
+As you can see we don't push the new message in our array on message creation.
```
Comment (scottinet) [inline doc/6/getting-started/vuejs/standalone/index.md]: why?

File: doc/6/getting-started/vuejs/standalone/index.md [hunk 11]
```diff
@@ -0,0 +1,118 @@
...
+Actually, We'll subscribe to the collection that contains our messages.
+So let's create our `subscribe_messages()` action. It will call the realtime controller of Kuzzle to allow us to [receive notifications](/sdk/js/6/controllers/realtime/subscribe/) on message creation:
```
Comment (scottinet) [inline doc/6/getting-started/vuejs/standalone/index.md]:
```suggestion
So let's create our `subscribe_messages()` action. It will call the Kuzzle's realtime controller to allow us to [receive notifications](/sdk/js/6/controllers/realtime/subscribe/) on message creations:
```

File: doc/6/getting-started/vuejs/standalone/index.md [hunk 12]
```diff
@@ -0,0 +1,118 @@
...
+
+<<< ./snippets/App.vue.snippet:11[js]
+
+To finish, just add an input field, bond to the `message` property and a button which calls our `sendMessage()` function:
```
Comment (scottinet) [inline doc/6/getting-started/vuejs/standalone/index.md]:
```suggestion
To finish, just add an input field bound to the `message` property, and a button which calls our `sendMessage()` function:
```

File: doc/6/getting-started/vuejs/standalone/index.md (hunk 11 above)
Comment (scottinet) [inline doc/6/getting-started/vuejs/standalone/index.md]: (nitpicking) wouldn't be better/more homogeneous to camelCase `subscribe_message`?

File: doc/6/getting-started/vuejs/standalone/index.md [hunk 13]
```diff
@@ -0,0 +1,118 @@
...
+
+<<< ./snippets/App.vue.snippet:12[html]
+
+And the following CSS class: 
+
+<<< ./snippets/App.vue.snippet:13[css]
+
+You can now add new messages to Kuzzle and receive the notification of the creation to update your state and display the new messages, enjoy :)
```
Comment (scottinet) [inline doc/6/getting-started/vuejs/standalone/index.md]: Stay formal:

```suggestion
You can now add new messages to Kuzzle and receive the creation notification to update your state and display the new messages.
```

File: doc/6/getting-started/vuejs/standalone/index.md [hunk 14]
```diff
@@ -0,0 +1,118 @@
...
+vue create kuzzle-playground
+```
+
+Install the kuzzle-sdk: 
+```bash
+cd kuzzle-playground
+npm install kuzzle-sdk
+```
+
+In the _App.vue_ file, you should remove the tag, the import and the component registration of the `HelloWorld` component, we won't use it.
```
[the remark this answers is not in the document]
Comment (berthieresteban) [reply doc/6/getting-started/vuejs/standalone/index.md]: I used backquotes for properties, method names and HTML tags. Should I do the opposite ?

File: doc/6/getting-started/.vuejs/standalone/src/App.vue [hunk 15]
```diff
@@ -0,0 +1,201 @@
...
+    },
+    /* snippet:end */
+    /* snippet:start:10 */
+    async sendMessage() {
+      if (this.message === "") return;
+      // Call the create method of the document controller
+      await kuzzle.document.create(
+        "chat",
+        "messages",
+        // Give as parameter the object that will be store in kuzzle
```
Comment (Aschen) [inline doc/6/getting-started/.vuejs/standalone/src/App.vue]: Nit/proposal
```suggestion
        // Pass the document to be stored in Kuzzle as a parameter
```

File: doc/6/getting-started/.vuejs/standalone/src/App.vue [hunk 16]
```diff
@@ -0,0 +1,201 @@
...
+    /* snippet:end */
+    /* snippet:start:11 */
+    async subscribeMessages() {
+      // Call the subscribe method of the realtime controller and receive the roomId
+      // Save the id of our subscription (we could need it to unsubscribe)
+      this.roomID = await kuzzle.realtime.subscribe(
+        "chat", // Id of the index
+        "messages", // Id of the collection
+        {}, // Filter
+        // Callback for notifications receive
```
Comment (Aschen) [inline doc/6/getting-started/.vuejs/standalone/src/App.vue]: Proposal/Nit
```suggestion
        // Callback to receive notifications
```

File: doc/6/getting-started/.vuejs/standalone/src/App.vue [hunk 17]
```diff
@@ -0,0 +1,201 @@
...
+        this.messages = [this.getMessage(hit), ...this.messages];
+      });
+    },
+    /* snippet:end */
+    /* snippet:start:2 */
+    async valid() {
+      // Etablish the connection
+      await kuzzle.connect();
+      // Check if 'chat' index exists
+      if (!(await kuzzle.index.exists("chat"))) {
```
Comment (Aschen) [inline doc/6/getting-started/.vuejs/standalone/src/App.vue]: Nit: unnecessary parenthesis
```suggestion
      if (! await kuzzle.index.exists("chat")) {
```

File: doc/6/version.md [hunk 18]
```diff
@@ -0,0 +1,9 @@
+---
+code: false
+type: page
+order: 0
+title: Kuzzle Documentation Version
+---
+
+
+<Version site="/sdk/js/6/" repository="https://github.com/kuzzleio/sdk-javascript.git" commit="8dd6cc7d416e4bd7e4ac93d405ff6fef22328af0" />
```
Comment (Aschen) [inline doc/6/version.md]: This file is obsolete and should be removed

File: doc/doc.sh [hunk 19]
```diff
@@ -20,9 +20,14 @@ fi
 
 case $1 in
   prepare)
-    echo "Clone documentation framework"
-    rm -rf framework
-    git clone --depth 10 --single-branch --branch master https://github.com/kuzzleio/documentation.git framework/
+    if [ -d "framework" ];
```
Comment (Aschen) [inline doc/doc.sh]: Incorrect merge, the previous version of the file should be keeped

File: doc/6/getting-started/.vuejs/cypress/support/commands.js [hunk 20]
```diff
@@ -0,0 +1,99 @@
+function reinitialisation() {
+  const kuzzle = Cypress.env('kuzzle');
+  
+  // Clear collection
+  return cy.request({
```
Comment (scottinet) [inline doc/6/getting-started/.vuejs/cypress/support/commands.js]: (nitpicking) better written like this:

```js
  // Clear collection
  return cy
    .request({
      url: `http://${kuzzle.host}:${kuzzle.port}/${kuzzle.index}/${kuzzle.collection}/_truncate`,
      method: 'DELETE',
    })
    .then(searchResponse => {
      cy.log(`Request : truncate ${kuzzle.collection} status : ${searchResponse.status}`);
      cy.wait(500);
    });
```

File: doc/6/getting-started/.vuejs/cypress/support/commands.js [hunk 21]
```diff
@@ -0,0 +1,99 @@
...
+      cy.wait(500);
+    });
+});
+
+
+Cypress.Commands.add('initialisation', () => {
+  const kuzzle = Cypress.env('kuzzle');
+
+  // Delete index if exists
+  cy.request({
```
Comment (scottinet) [inline doc/6/getting-started/.vuejs/cypress/support/commands.js]: Promises must either be returned or have their own `catch` handler.

(this remark must be applied to all created promises)

File: doc/6/getting-started/.vuejs/cypress/support/commands.js [hunk 22]
```diff
@@ -0,0 +1,99 @@
...
+            cy.log(`Request : delete ${kuzzle.index} status : ${deleteResponse.status}`);
+          });
+      }
+    }).then(() => {
+      // Create index
+      cy.request({
+        url: `http://${kuzzle.host}:${kuzzle.port}/${kuzzle.index}/_create`,
+        method: 'POST',
+      })
+        .then(createResponse => {
```
Comment (scottinet) [inline doc/6/getting-started/.vuejs/cypress/support/commands.js]: This promise chain should be flattened.

(this remark applies to other promise subchains that can/should be flattened)

File: doc/6/getting-started/vuejs/standalone/index.md [hunk 23]
```diff
@@ -0,0 +1,121 @@
...
+
+We need to write a simple method that will [create](/sdk/js/6/controllers/document/create/) a new message document in Kuzzle.
+
+<<< ./snippets/App.vue.snippet:10[js]
+
+As you can see we don't push the new message in our array on message creation.
+
+Indeed, we will receive notifications from Kuzzle each time we modify our message collection (even if it is a message creation on our part) that we will use to add the messages in our array.
+
+Actually, we need to subscribe to the collection that contains our messages.
```
Comment (scottinet) [inline doc/6/getting-started/vuejs/standalone/index.md]: False friend.

```suggestion
Now, we need to subscribe to the collection that contains our messages.
```

**Comment by scottinet** [comment]
Cypress tests still failing.
